torch/csrc/jit/script/module.h

```diff
@@ -481,6 +481,25 @@ struct Module {
     }
     fn(*this);
   }
+  void train(bool on = true) {
+    for (auto& submod : get_modules()) {
+      submod.value().module->train(on);
+    }
+    auto t = autograd::make_variable(at::full({}, on ? 1 : 0, at::kLong));
+    register_parameter("training", std::move(t), true);
+  }
+  void eval() {
+    train(/*on=*/false);
+  }
+  bool is_training() {
+    if (auto p = find_parameter("training")) {
+      return (*p->slot()).item().toLong() == 1;
```

Comment:
Here you are converting the tensor first to a `Scalar`, and then to a long. You can just use `item<T>` to get the value of the tensor without the intermediate `Scalar`:
```
(*p->slot()).item<int64_t>()
```